home/tests/test_signup.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+from django.contrib.auth.models import User
+from home.models import Profile
+from pytest_django.asserts import assertTemplateUsed
+from home.forms import SignUpForm
+
+
+@pytest.mark.django_db
+class TestSignUp:
+
+    @pytest.fixture
+    def valid_signup_details(self):
+        return {'username': 'username', 'email': 'user@e.com', 'gender': 'U',
+                'password1': 'ido123123', 'password2': 'ido123123'}
+
+    def get_signup_page_status_code(self, client):
+        response = client.get('/signup')
+        assert response.status_code == 200
+
+    def test_valid_signup(self, valid_signup_details, client):
+        client.post('/signup/', data=valid_signup_details)
+        try:
+            user = User.objects.filter(username=valid_signup_details['username'])
+            assert user is not None
+            profile = Profile.objects.filter(user=user)
+            assert profile is not None
+        except ValueError:
```

Comment:
You should avoid handling the exception like this, you are effectively hiding useful details from the test results report.

Reply:
fixed